.github/workflows/gradle.yml

```
@@ -0,0 +1,31 @@
+name: Automatic Testing - Gradle
+
+#The action will be triggered everytime a pull request is made that includes files in the src directory (master branch)
+on: 
+  pull_request:
+    branches: 
+      - master
+    paths:
+      - 'src/**'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: windows-latest
```
We are mainly using windows now, later we are going to add Linux as well

---

Yes, but this is only the GitHub hosted virtual machine in which the action gets executed. (I mean it doesn't influence our development at all) so  I thought of going with the default configuration. Do we really need to differentiate here?

---

normally not, but if we do something platform dependent (like network sockets), then we will need to differentiate. Anyway we can keep it as it is now and see if it causes problems in the future.